docs/architecture/adr-056-proving-amnesia-attack.md

```diff
@@ -0,0 +1,85 @@
+# ADR 56: Proving Amnesia Attack
+
+## Changelog
+
+- 02.04.20: Initial Draft
+
+## Context
+
+Whilst most created evidence of malicious behaviour is self evident such that any individual can verify them independently there are types of evidence that require collaboration from the network of validators, known as global evidence, in order to accumulate enough information to create evidence that is individually verifiable and can therefore be processed through consensus. Fork Accountability as a whole constitutes of detection, proving and punishing. This ADR addresses how to prove infringement from global evidence.
+
+So far, only [flip flopping](https://github.com/tendermint/spec/blob/master/spec/consensus/light-client/accountability.md#flip-flopping) attacks such as amnesia (which will be what the reactor initially focuses on) and back to the past (only causes light forks) have been identified as forms of global evidence that require more information from other validators.
+
+## Decision
+
+A new reactor will perform the task of turning global evidence to individual evidence. This special circumstance reactor could be combined with the current evidence reactor although the evidence reactor is currently used for gossiping and storing of evidence and so it's purpose would expand. The reactor is initialized when a full node receives PotentialAmnesiaEvidence from a light client.
+
+*REMARK: This means that only the set of full nodes that are directly connected to the light client that observed the misbehaviour will process the evidence. Is this sufficient, do we want to gossip to further nodes, do we want just a single node to process evidence to avoid too much communication overhead?*
```

Comment:
I think we want to gossip PotentialAmnesiaEvidence to as many nodes as we can, so they halt

Reply:
Yes I completely agree. I was just thinking of having a mechanism that ensures that not every node runs this reactor at the same time.